core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/ExecutionInterceptor.java

```diff
@@ -219,12 +238,49 @@ default void afterTransmission(Context.AfterTransmission context, ExecutionAttri
      *
      * @param context The current state of the execution, including the SDK and HTTP requests and the current HTTP response.
      * @param executionAttributes A mutable set of attributes scoped to one specific request/response cycle that can be used to
-     *                            give data to future lifecycle methods.
+     * give data to future lifecycle methods.
      * @return The potentially-modified HTTP response that should be given to the unmarshaller. Must not be null.
      */
-    default SdkHttpFullResponse modifyHttpResponse(Context.ModifyHttpResponse context,
+    default SdkHttpFullResponse modifyHttpResponse(SdkHttpFullResponse toModify,
+                                                   Context.ModifyHttpResponse context,
                                                    ExecutionAttributes executionAttributes) {
-        return context.httpResponse();
+        return toModify;
+    }
+
+    /**
+     * Modify the {@link SdkHttpFullRequest} before it is unmarshalled into an {@link SdkResponse}.
+     *
+     * <p>Note: Unlike many other lifecycle methods, this one may be invoked multiple times. If the {@link RetryPolicy} determines
+     * the error code returned by the service is retriable, this will be invoked for each response returned by the service.
+     *
+     * @param context The current state of the execution, including the SDK and HTTP requests and the current HTTP response.
+     * @param executionAttributes A mutable set of attributes scoped to one specific request/response cycle that can be used to
+     * give data to future lifecycle methods.
+     * @return The potentially-modified HTTP response that should be given to the unmarshaller. Must not be null.
+     */
+    default Publisher<ByteBuffer> modifyAsyncHttpResponseContent(Publisher<ByteBuffer> toModify,
+                                                                 Context.ModifyAsyncHttpResponseContent context,
+                                                                 ExecutionAttributes executionAttributes) {
```

Comment:
The values to be modified aren't usually passed separately. Any reason we're changing that just for these?